troubleshooting/scripts/kafka_crd_fix.sh

```diff
@@ -146,9 +207,40 @@ remove_old_crd_versions() {
   done
 }
 
-echo "Checking for old kafka CRD versions"
+cleanup() {
+  # Remove last-applied-configuration in annotations and also preserveUnknownFields field in spec
+  CRDS=$(kubectl get crds -lapp=strimzi -o jsonpath='{.items[*].metadata.name}')
+
+  cat << EOF
+Removing
+    metadata.annotations.last-applied-configuration
+    spec.preserveUnknownFields
+from kafka CRDs
+EOF
+
+  for crd in ${CRDS}; do
+    crd_json=$(kubectl get crd "${crd}" -o json)
+
+    has_last_applied=$(echo "$crd_json" | jq -e '.metadata.annotations["kubectl.kubernetes.io/last-applied-configuration"]' > /dev/null && echo true || echo false)
+    has_preserve_unknown=$(echo "$crd_json" | jq -e 'has("spec") and .spec | has("preserveUnknownFields")' > /dev/null && echo true || echo false)
+    if [[ $has_last_applied == "true" ]]; then
+      echo "Removing last-applied-configuration from CRD ${crd}"
+      kubectl patch --type merge crd "${crd}" -p '{"metadata":{"annotations":{"kubectl.kubernetes.io/last-applied-configuration": null}}}'
+    fi
+
+    if [[ $has_preserve_unknown == "true" ]]; then
+      echo "Setting preserveUnknownFields to null in CRD ${crd}"
+      kubectl patch --type merge crd "${crd}" -p '{"spec":{"preserveUnknownFields": null}}'
+    fi
```

Comment:
This works, but it's a lot of extra work to first check individually for both items, and then patch separately for both items.  It can all be done in a single `kubectl patch` command without any pre-checks, which will still return success even if it doesn't change anything.  And the advantage of just applying it to everything is then you get a list of both what was changed and what wasn't changed.

Reply:
It's okay as it is, a few extra commands isn't a big deal.